### DISCUSSION TITLE
Tasks, Vendors & Violations

### DISCUSSION THREAD

**Discussion by fkhaja** [microrealestate/microrealestate#80] [Ideas]
I think it would be great to add property specific tasks, assign them to vendors and manage property, company and landlord level tasks in a separate dashboard. Thoughts/opinions?

**Comment by camelaissani** [comment]
Can you elaborate more please?
Do you mean to have 2 profiles with different rights:
* admin/landlord: can access the full app (all pages)
* vendors/renters: can create tenant leases, enter the payments, terminate leases and then access the tenant page and the rent page

**Reply by fkhaja** [reply]
Hi Camel,
I'm thinking about the ability to track maintenance requests, including their time and cost. These maintenance requests (or tasks) will need to be assigned to a person/company (vendors) and the status will be visible to both tenants and landlords.

User profiles can be as follows:
- admins/managers:  access full app (restricted view, eg. retract accounting reports if required). Ideally, the profiles should be customizable based on permissions
- landlords: full access to the properties they own
- vendors: access open tasks assigned to them
- renters: create maintenance tickets, see payment history, view contracts, etc

The managers should be able to see all tasks open in the system
Landlords would see the tasks related to their properties
Vendors will only see tasks assigned to them

Let me know if anything doesn't make sense.
